relayer-cli/src/commands/query/channels.rs

```diff
@@ -1,60 +1,184 @@
+
+        match &cmd.destination_chain {
+            Some(dst_chain_id) if dst_chain_id != &counterparty_chain_id => {
+                continue;
+            }
+            _ => { /* proceed */ }
+        }
+
+        if !cmd.verbose {
+            output.push(PortChannelId {
+                channel_id,
+                port_id,
+            });
+        } else {
+            let channel_counterparty = channel_end.counterparty().clone();
+            let connection_counterparty = connection_end.counterparty().clone();
+            let counterparty_client_id = connection_counterparty.client_id().clone();
+
+            let counterparty_connection_id = connection_counterparty.connection_id.ok_or_else(|| {
+                format!(
+                    "connection end for {} on chain {} @ {:?} does not have counterparty connection id: {:?}",
+                    connection_id,
+                    chain_id,
+                    chain_height,
+                    connection_end
+                )
+            })?;
+
+            let counterparty_port_id = channel_counterparty.port_id().clone();
+
+            let counterparty_channel_id = channel_counterparty.channel_id.ok_or_else(|| {
+                format!(
+                    "channel end for {}/{} on chain {} @ {:?} does not have counterparty channel id: {:?}",
+                    port_id, channel_id, chain_id, chain_height, channel_end
+                )
+            })?;
+
+            let chain_config_b = config.find_chain(&counterparty_chain_id).ok_or_else(|| {
+                format!(
+                    "counterparty chain '{}' not found in configuration file",
+                    counterparty_chain_id
+                )
+            })?;
+
+            let counterparty_chain = CosmosSdkChain::bootstrap(chain_config_b.clone(), rt.clone())?;
+            let counterparty_chain_height = counterparty_chain.query_latest_height()?;
+            let counterparty_connection_end = counterparty_chain
+                .query_connection(&counterparty_connection_id, counterparty_chain_height)?;
+            let counterparty_client_state = counterparty_chain
+                .query_client_state(&counterparty_client_id, counterparty_chain_height)?;
+            let counterparty_channel_end = counterparty_chain.query_channel(
+                &counterparty_port_id,
+                &counterparty_channel_id,
+                counterparty_chain_height,
+            )?;
+
+            output_verbose.push(ChannelEnds {
+                channel_end,
+                connection_end,
+                client_state,
+
+                counterparty_channel_end,
+                counterparty_connection_end,
+                counterparty_client_state,
+            });
+        }
+    }
+
+    let output = if !cmd.verbose {
+        QueryChannelsOutput::Summary(output)
+    } else {
+        QueryChannelsOutput::Verbose(output_verbose)
+    };
+    Ok(output)
```

Comment:
We can do `Output::success` here to skip having to define `QueryChannelsOutput`.

Reply:
I actually like having a well-defined return type for the actual command that is only converted to an `Output` at the top-level. Makes the code more explicit and easier to read and reason about imho.

Reply:
Sounds good. I'm fine either way.